learner/sampler/slurper.py

```diff
@@ -0,0 +1,134 @@
+from itertools import islice
+import math
+import os
+import random
+import sys
+import time
+from typing import List
+
+##TO DO:
+#Only call this when the file is of more than ~1/3 total available memory size of the computer. (or approaching max of python 2)
+#Ensure this works for unstructured file types - write this for xml and json
+#Catch encoding errors inside the functions (get filesize)
+#Connect to the API for the rest of the project
+
+class Slurper(object):
+    def __init__(self, filename: str, encoding: str = 'UTF-8') -> None:
+        self.file = filename
+        self.encoding = check_file_encodings()
+        self.nlines = estimate_line_size()
+        self.sample_size = get_sample_size()
+    
+    def check_file_encodings(file: str = None, encoding: str = None) -> str:
+        """Ensure that the file can be opened by cycling through common file types."""        
+        if not file:
+            file = self.filename
+        if not encoding:
+            encoding = self.encoding            
```

Comment:
From the way that it's been called, it looks like `self.encoding` as not been set yet. This might(?) lead to a recursive loop, given that `self.encoding` is currently a function call. That's what it feels like rn anyway. Unsure what kind of behavior this will exhibit, but since `self.encoding` has no value yet, feels like this might be a useless line.